cpp/ql/src/semmle/code/cpp/models/implementations/MemberFunction.qll

```diff
@@ -35,7 +35,11 @@ class ConversionConstructorModel extends Constructor, TaintFunction {
 class CopyConstructorModel extends CopyConstructor, DataFlowFunction {
   override predicate hasDataFlow(FunctionInput input, FunctionOutput output) {
     // data flow from the first constructor argument to the returned object
-    input.isParameter(0) and
+    (
+      input.isParameter(0)
+      or
+      input.isParameterDeref(0)
+    ) and
```

Comment:
Surely `input.isParameter(0)` was wrong to begin with, and one of the data-flow libraries should be adjusted so we only have to write `input.isParameterDeref(0)` here. Or? Not that it has to happen in this PR.

Cc @geoffw0

Reply:
I agree, we should just delete `input.isParameter(0)` (or if that breaks any of the AST tests then ask me to do it in a follow-up PR instead).

Reply:
Opened https://github.com/github/codeql/pull/4642 .